R2.14 documentation updates

inc1t3Ful opened the pull request:
** work in progress
- still missing the FeatureCompatibility Chart

inc1t3Ful committed:
Update all data source docs to standardize all formatting

** standardize formatting
** fix grammar
** fix incorrect descriptions
++ add proper data source descriptions

inc1t3Ful committed:
Update docs and standardize formatting for new resources in R2.14

** tgw_peering
** tgw_peering_domain_conn
** transit_external_device_conn

inc1t3Ful committed:
Added Release Notes, updated changelist and missing reference for gateway changes

inc1t3Ful committed:
(13570) Note added about GeoVPN and gateway VPN config management

zhiqiangjin reviewed (commented):
data source

zhiqiangjin commented on website/docs/d/aviatrix_firenet.html.markdown:
```diff
@@ -2,12 +2,14 @@
-Use this data source to get the Aviatrix firenet for use in other resources.
+The **aviatrix_firenet** data source provides details about a specific FireNet created by the Aviatrix Controller.
+
+This resource can prove useful when a module accepts a FireNet's detail as an input variable.
```
"This data source"

inc1t3Ful replied:
I used the formatting you see because I wanted to keep consistent formatting across all docs, similar to our resource markdowns. What do you think?

By avoiding "This data source", and substituting with bolded text, it makes it clear what the sentence that proceeds it talks about. Formatting was modeled after AWS's.

zhiqiangjin replied:
I mean in line 12, "This resource" should be "This data source"

inc1t3Ful replied:
Ah I see now. Will change

zhiqiangjin commented on website/docs/d/aviatrix_gateway.html.markdown:
```diff
@@ -2,12 +2,14 @@
-Use this data source to get the Aviatrix gateway for use in other resources.
+The **aviatrix_gateway** data source provides details about a specific gateway created by the Aviatrix Controller.
+
+This resource can prove useful when a module accepts a gateway's detail as an input variable. For example, requiring the gateway's name configuring a site2cloud connection.
```
"This data source"?

zhiqiangjin commented on website/docs/d/aviatrix_spoke_gateway.html.markdown:
```diff
@@ -2,12 +2,14 @@
-Use this data source to get the Aviatrix spoke gateway for use in other resources.
+The **aviatrix_spoke_gateway** data source provides details about a specific spoke gateway created by the Aviatrix Controller.
+
+This resource can prove useful when a module accepts a spoke gateway's detail as an input variable.
```
Same

zhiqiangjin commented on website/docs/d/aviatrix_transit_gateway.html.markdown:
```diff
@@ -2,12 +2,14 @@
-Use this data source to get the Aviatrix transit gateway for use in other resources.
+The **aviatrix_transit_gateway** data source provides details about a specific transit gateway created by the Aviatrix Controller.
+
+This resource can prove useful when a module accepts a transit gateway's detail as an input variable. For example, requiring the transit gateway's name for a spoke gateway's attachment.
```
"This data source"

zhiqiangjin commented on website/docs/d/aviatrix_vpc.html.markdown:
```diff
@@ -2,17 +2,19 @@
-Use this data source to get the Aviatrix vpc for use in other resources.
+The **aviatrix_vpc** data source provides details about a specific VPC created by the Aviatrix Controller.
+
+This resource can prove useful when a module accepts any form of VPC detail as an input variable. For example, requiring a subnet CIDR specification when creating a gateway.
```
"This data source"

inc1t3Ful committed:
Fix mis-matched Release Notes

inc1t3Ful committed:
Update v2 upgrade guide to align with R2.14

inc1t3Ful committed:
Update resource docs with embedded links to feature docs

++ added links to main Aviatrix site for feature details
** improved titles and added explicit details, changed wordings
** fixed formatting on v2 upgrade guide

inc1t3Ful committed:
Fix incorrect naming of data_source

** resource -> data source

inc1t3Ful committed:
Update FeatureCompatibility sheet
